routing/index_road_graph.cpp

```diff
@@ -78,8 +85,14 @@ void IndexRoadGraph::GetEdges(Junction const & junction, bool isOutgoing, TEdgeV
     if (IndexGraphStarter::IsFakeSegment(segment))
       continue;
 
-    edges.emplace_back(FeatureID(m_mwmId, segment.GetFeatureId()), segment.IsForward(),
-                       segment.GetSegmentIdx(), GetJunction(segment, false /* front */),
+    platform::CountryFile const & file = m_numMwmIds->GetFile(segment.GetMwmId());
+    MwmSet::MwmHandle handle = m_index.GetMwmHandleByCountryFile(file);
```

Comment:
This is VERY expensive, as requires mutex lock/unlock. Do you really want this code be executed in a loop? On each GetEdges() call?